mteb/evaluation/evaluators/InstructionRetrievalEvaluator.py

```diff
@@ -0,0 +1,263 @@
+import logging
+import heapq
+from typing import Dict, List, Tuple
+import pandas as pd
+import tqdm
+
+import pytrec_eval
+from sentence_transformers import SentenceTransformer
+from sentence_transformers.models import Transformer, WordEmbeddings
+import torch
+from collections import defaultdict
+
+from .Evaluator import Evaluator
+from .utils import cos_sim, dot_score, mrr, recall_cap, hole, top_k_accuracy
+
+logger = logging.getLogger(__name__)
+
+# Adapted from https://github.com/beir-cellar/beir/blob/f062f038c4bfd19a8ca942a9910b1e0d218759d4/beir/retrieval/search/dense/exact_search.py#L12
+class DenseRetrievalExactSearch:    
```

Comment:
Unfortunately, I had to modify this as well --- we need some way of caching the document embeddings.  I could move this class to AbstractRetrieval with the new additions?

Reply:
Ah sorry - meant add to existing